jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/WarTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.Command;
+import com.google.cloud.tools.jib.IntegrationTestingConfiguration;
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+/** Integration tests for building WAR images. */
+public class WarTest {
```

Comment:
`WarIntegrationTest`? I think this should just belong in `JibPluginIntegrationTest` since the tests in there are for non-wars. Or, we can set a convention of naming like `JibPluginIntegrationTestWar.java` to indicate this class is just an extension onto integration tests for `JibPlugin`.

Reply:
I'm not really getting the part "integration tests for `JibPlugin`." The test looks more like an end-to-end test, actually running a Gradle build (albeit programmatically) with

```java
GradleRunner.create().withProjectDir(projectRoot.toFile()).withPluginClasspath()
```

and not like you directly instantiate `JibPlugin` or testing each method in `JibPlugin`.

BTW, I think `WarIntegrationTest` sounds good. I plan to add many tests, so it doesn't sound bad to classify tests with more classes. I think we don't need to put everything into a single class.

Reply:
The `JibPlugin` class is the class that is directly applied to the Gradle build script and executed on the project build. The `JibPluginIntegrationTest` is testing the integration of `JibPlugin` with external dependencies (like GCR). I think if we break out `JibPluginIntegrationTest` into multiple classes, we should group them by the test project, so we'd have something like `SimpleProjectIntegrationTest`, `MultiProjectIntegrationTest`, `EmptyProjectIntegrationTest`, and `DefaultTargetProjectIntegrationTest`, and `WarProjectIntegrationTest`.

Reply:
OK, will try breaking up `JibPluginIntegrationTest`.

Reply:
Can probably leave that to a follow-up PR too.

Reply:
OK, I'm almost done with breaking it up, but I'll actually put that up in a later PR.